src/state_types/models/ctx/update_profile.rs

```diff
@@ -96,6 +96,14 @@ pub fn update_profile<Env: Environment + 'static>(
                     .iter()
                     .map(|addon| &addon.transport_url)
                     .position(|transport_url| *transport_url == addon.transport_url);
+                let push_to_api_effects = match &profile.auth {
```

Comment:
that block needs to be moved below the next block because that way an older version of profile.addons will be pushed to API